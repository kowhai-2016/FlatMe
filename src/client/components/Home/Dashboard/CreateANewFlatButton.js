import React from 'react'

const CreateANewFlat = (props) => {
  return (
    <div className='row'>
      <button onClick={props.open} className='btn btn-default glyphicon glyphicon-plus'> create a new flat</button>
    </div>
  )
}

export default CreateANewFlat
