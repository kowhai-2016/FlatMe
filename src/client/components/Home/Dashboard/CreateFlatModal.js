import React, { PropTypes } from 'react'
import { Modal, Button } from 'react-bootstrap'
import { Control, Errors, Form } from 'react-redux-form'

const CreateFlatModal = props => {
  return (
    <Modal class='CreateFlatModal' show={props.show} onHide={props.close}>
      <Form model='forms.newFlat' onSubmit={props.onSubmit}>
        <Modal.Header closeButton>
          <Modal.Title>Create a new Flat</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div>
            <div>
              <label>Flat Name:</label>
              <Control model='.flatName' className='modal-input' />
            </div>
          </div>
          <Errors model='forms.newFlat' />
        </Modal.Body>
        <Modal.Footer>
          <Button type='submit'>Virtually create your flat</Button>
        </Modal.Footer>
      </Form>
    </Modal>
  )
}

CreateFlatModal.propTypes = {
  close: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired
}

export default CreateFlatModal
