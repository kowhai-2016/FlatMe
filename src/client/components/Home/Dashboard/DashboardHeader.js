import React from 'react'

const DashboardHeader = () => {
  return (
    <div className='row'>
      <div className='col-md-12'>
        <h1 id='myflats'>My Flats</h1>
      </div>
    </div>
  )
}

export default DashboardHeader
