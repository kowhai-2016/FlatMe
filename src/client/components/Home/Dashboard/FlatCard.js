import React, { PropTypes } from 'react'
import { Link } from 'react-router'

const FlatCard = React.createClass({
  render () {
    const flats = this.props.flats.map((flat, index) => {
      return (
        <div className='row' key={index}>
          <div className='col-md-6 col-md-offset-3'>
            <div className='panel-group main-flat-card'>
              <div className='panel panel-default flat-card'>
                <div className='panel-heading'><h4>My Flat # {index + 1}</h4></div>
                <div className='panel-body'>
                  <ul>
                    <li>
                      <Link to={`/flat/${flat.id}`}>
                        <h3><img src='/images/livingroom-window-with-curtains.svg' className='curtains' />{flat.flatName}</h3>
                      </Link>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      )
    })
    return (
      <div>
        {flats}
      </div>
    )
  }
})

FlatCard.propTypes = {
  flats: PropTypes.arrayOf(
    PropTypes.shape({
      flatName: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired
    })
  ).isRequired
}

export default FlatCard
