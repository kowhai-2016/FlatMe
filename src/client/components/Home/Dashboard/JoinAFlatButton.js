import React from 'react'

const JoinAFlat = (props) => {
  return (
    <div className='row'>
      <button onClick={props.open} className='btn btn-default glyphicon glyphicon-ok'> join a flat</button>
    </div>
  )
}

export default JoinAFlat
