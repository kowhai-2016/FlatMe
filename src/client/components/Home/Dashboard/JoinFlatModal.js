import React, { PropTypes } from 'react'
import { Modal, Button } from 'react-bootstrap'
import { Control, Errors, Form } from 'react-redux-form'

const JoinFlatModal = props => {
  const onSubmit = values => {
    props.onSubmit(values)
    props.close()
  }
  return (
    <Modal className='JoinFlatModal' show={props.show} onHide={props.close}>
      <Form model='forms.joinFlat' onSubmit={onSubmit}>
        <Modal.Header closeButton>
          <Modal.Title>Join a Flat</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <div>
            <div className='NameInput'>
              <label>Search by Flat Name: </label>
              <Control model='.joinFlatName' className='modal-input' />
            </div>
          </div>
          <Errors model='forms.joinFlat' />
        </Modal.Body>
        <Modal.Footer>
          <Button type='submit'>Join my new flat</Button>
        </Modal.Footer>
      </Form>
    </Modal>
  )
}

JoinFlatModal.propTypes = {
  close: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  show: PropTypes.bool.isRequired
}

export default JoinFlatModal
