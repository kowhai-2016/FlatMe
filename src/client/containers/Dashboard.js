import { connect } from 'react-redux'

import Dashboard from '../components/Home/Dashboard'

import { createNewFlat, fetchUserFlats, joinFlat } from '../actions'

const mapStateToProps = state => {
  return {
    user: state.account.user
  }
}

const mapDispatchToProps = (dispatch, state) => {
  return {
    createNewFlat: (flat, user) => dispatch(createNewFlat(flat, user)),
    fetchUserFlats: id => dispatch(fetchUserFlats(id)),
    joinFlat: ({ joinFlatName }) => dispatch(joinFlat(joinFlatName))
  }
}
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Dashboard)
