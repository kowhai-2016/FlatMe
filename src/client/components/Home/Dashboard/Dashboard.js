import React, { PropTypes } from 'react'

import CreateFlatModal from './CreateFlatModal'
import DashboardHeader from './DashboardHeader'
import CreateANewFlat from './CreateANewFlatButton'
import JoinAFlat from './JoinAFlatButton'
import JoinFlatModal from './JoinFlatModal'
import FlatCard from './FlatCard'
import './style.css'

const Dashboard = React.createClass({

  componentDidMount () {
    this.props.fetchUserFlats(this.props.user.id)
  },

  closeCreate () {
    this.setState({
      showCreate: false
    })
  },

  closeJoin () {
    this.setState({
      showJoin: false
    })
  },

  getInitialState () {
    return {
      show: false,
      showCreate: false,
      showJoin: false
    }
  },

  openCreate () {
    this.setState({
      showCreate: true
    })
  },

  openJoin () {
    this.setState({
      showJoin: true
    })
  },
  handleSubmit (values) {
    this.props.createNewFlat(values, this.props.user)
  },
  render () {
    const flats = this.props.user.flats ? this.props.user.flats : []
    return (
      <div className='Dashboard container-fluid'>
        <div className='col-md-12 bg-flat'>
          <div className='container containerFlat'>
            <div className='row buttons-container'>
              <div className='col-md-3 create-flat'>
                <CreateANewFlat open={this.openCreate} />
              </div>
              <div className='col-md-3'>
                <JoinAFlat open={this.openJoin} />
              </div>
            </div>
            <DashboardHeader />
            <FlatCard flats={flats} />
            <CreateFlatModal
              close={this.closeCreate}
              onSubmit={this.handleSubmit}
              show={this.state.showCreate}
              />
            <JoinFlatModal
              close={this.closeJoin}
              onSubmit={this.props.joinFlat}
              show={this.state.showJoin}
              />
          </div>
        </div>
      </div>
    )
  }
})

Dashboard.propTypes = {
  createNewFlat: PropTypes.func.isRequired,
  fetchUserFlats: PropTypes.func.isRequired,
  user: PropTypes.shape({
    flats: PropTypes.arrayOf(PropTypes.object)
  }).isRequired,
  joinFlat: PropTypes.func.isRequired
}

export default Dashboard
