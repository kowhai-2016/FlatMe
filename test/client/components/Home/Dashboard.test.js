/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Dashboard from '../../../../src/client/components/Home/Dashboard'

test('Dashboard renders correctly', () => {
  const createNewFlat = () => {}
  const fetchUserFlats = () => {}
  const joinFlat = () => {}
  const user = {}
  const tree = renderer.create(
    <Dashboard
      createNewFlat={createNewFlat}
      fetchUserFlats={fetchUserFlats}
      joinFlat={joinFlat}
      user={user}
      />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
