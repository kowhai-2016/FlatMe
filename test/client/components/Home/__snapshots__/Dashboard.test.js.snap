exports[`test Dashboard renders correctly 1`] = `
<div
  className="Dashboard container">
  <div
    className="row buttons-container">
    <div
      className="col-md-3 create-flat">
      <div
        className="row">
        <button
          className="btn btn-default glyphicon glyphicon-plus flat-button"
          onClick={[Function]}>
           create a new flat
        </button>
      </div>
    </div>
    <div
      className="col-md-3">
      <div
        className="row">
        <button
          className="btn btn-default glyphicon glyphicon-ok flat-button"
          onClick={[Function]}>
           join a flat
        </button>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-12">
      <h1
        id="myflats">
        My Flats
      </h1>
    </div>
  </div>
  <div />
</div>
`;
